### PR TITLE
Data Explorer: Revamped get_data_values RPC and stub out as-yet-unsupported set_column_filters RPC 

### PR DESCRIPTION
**Pull request by wesm**
As agreed to with @softwarenerd to align with the refactoring he's been doing in the data plane for the data explorer:

* There are instances where it is low-friction for the frontend to request an array of indices that it doesn't have cached for a particular column, whether other times it may be easier to request a range. 
* Similarly, what needs to be requested for each column may vary as the user moves around the table, so it is useful to be able to request a different range or set of indices for each column in the get data values request
* Accordingly, I've introduced a new "column_selection" OpenRPC type and an "array_selection" (range or indices) union to allow this to be expressed more exactly
* Relatedly, in light of the above, it doesn't make sense to always return the row labels in the get_data_values RPC request, so this adds as `has_row_labels` flag to the `get_state` response and a new `get_row_labels` request so these can be retrieved separately.
* I renamed a few things for clarity while we are in the process of breaking APIs

Lastly, in this PR I stubbed out placeholders (including in the state updates and features) for the `set_column_filters` request which can be implemented in a follow up PR. 

### QA Notes

We only need to verify that the existing smoke and CI tests continue to pass.

### CODE REVIEW

**Comment by wesm** [comment]
@dfalbel -- take a look at the protocol changes and let me know if you foresee any issues. I think this should be relatively easy to integrate into Ark -- I was going to try to do it myself but I figure you would be faster =)

**Review by dfalbel — DISMISSED** [review]
I'm OK with the protocol changes. The only small concern is that the GetDataValues change can affect performance as we will no longer be able to use an optimization that allows us to pre-filter all columns using the same selection.

File: extensions/positron-python/python_files/positron/positron_ipykernel/data_explorer_comm.py [hunk 1]
```diff
@@ -325,8 +332,13 @@ class TableData(BaseModel):
...
 
-    row_labels: Optional[List[List[StrictStr]]] = Field(
-        default=None,
+
+class TableRowLabels(BaseModel):
+    """
+    Formatted table row labels formatted as strings
+    """
+
+    row_labels: List[List[StrictStr]] = Field(
```
Comment (dfalbel) [inline extensions/positron-python/python_files/positron/positron_ipykernel/data_explorer_comm.py:341]: Do we really this to be a list of lists of row labels? I'd expect it to be `List[StrictStr]`, an array of strings for the selected region.

Comment (wesm) [reply extensions/positron-python/python_files/positron/positron_ipykernel/data_explorer_comm.py:341]: pandas has MultiIndexes where you can use multiple layers of row labels, so this is future proofing for when the front end can handle that properly

**Comment by wesm** [comment]
> The only small concern is that the GetDataValues change can affect performance as we will no longer be able to use an optimization that allows us to pre-filter all columns using the same selection.

I'd be interested to measure the difference -- seems like the difference (slicing/indexing one column at a time vs. slicing the whole table) should not be too great?

**Pull request merged**